### ISSUE TITLE
[BUG] Serializing StringColumn's slows when offset encountered

### ISSUE THREAD

**Issue by jakirkham** [rapidsai/cudf#4991]
**Describe the bug**

When serializing a `StringColumn` with an `offsets`, am noticing a fair bit of time is spent adjusting for the offset. In particular this line shows up during profiling. Though the lines after are likely also expensive.

https://github.com/rapidsai/cudf/blob/65bbee7920182f709a269b8f0aba4b2434f4beb3/python/cudf/cudf/core/column/string.py#L1906

Digging a bit deeper it appears a fair bit of time is spent here where a device-to-host transfer occurs. Though likely a host-to-device transfer is needed for later operations. So avoid these transfers would like help, but there may be other ways to pursue this.

https://github.com/rapidsai/cudf/blob/65bbee7920182f709a269b8f0aba4b2434f4beb3/python/cudf/cudf/core/column/column.py#L417

**Steps/Code to reproduce bug**

```python
import cudf

s = cudf.Series(["abc", "def", None, "ghi"])
s = s[1:]

s.serialize()
```

**Expected behavior**

**Environment overview (please complete the following information)**
 - Environment location: DGX-1 or DGX-2
 - Method of cuDF install: Conda

**Environment details**
Standard cuDF install with Conda.

**Additional context**
NA

cc @VibhuJawa @rgsl888prabhu @kkraus14

**Comment by kkraus14** [comment]
FYI just to give some context here, this is only expensive for sending sliced columns where we need to grab the scalar value of the offset based on the slice, and run a binaryop against a device scalar constructed from that scalar.

I believe there's ongoing work to allow building a device scalar without having to do a D --> H copy or synchronizing the stream here: #4900

**Comment by jakirkham** [comment]
Awesome! Thanks for sharing Keith 😄

**Comment by rgsl888prabhu** [comment]
@kkraus14 but you would need to do D->H at the end while creating the char column , [to get size of the char column](https://github.com/rapidsai/cudf/blob/65bbee7920182f709a269b8f0aba4b2434f4beb3/python/cudf/cudf/core/column/string.py#L1913) and [to offset the start of char column](https://github.com/rapidsai/cudf/blob/65bbee7920182f709a269b8f0aba4b2434f4beb3/python/cudf/cudf/core/column/string.py#L1919)

**Comment by karthikeyann** [comment]
> I believe there's ongoing work to allow building a device scalar without having to do a D --> H copy or synchronizing the stream here: #4900

D --> H copy happens for string_view and dictionary columns in `get_element` #4900 
For fixed_width scalars, there is no D --> H copy.
In all cases, there is a kernel launch (to copy both data and validity).

**Comment by kkraus14** [comment]
These are always integer columns, so that kernel launch should be a lot cheaper than a D --> H --> D that we currently have 😄.

Getting the size of the char column is still going to require a D --> H though.

**Comment by rgsl888prabhu** [comment]
Tested locally on q10 with latest cudf build as of now and along with PR #4900, the time consumption reduces from 22.42 s to 11.03 seconds. I have attached the dask profiling just for reference.

Note: This was not on whole set-up with 136 workers, but simulated 136 workers on desktop.


[profiled-main.zip](https://github.com/rapidsai/cudf/files/4561683/profiled-main.zip)

**Comment by rgsl888prabhu** [comment]
Closing this as we have merged #5072
